### PR TITLE
Add bun support

### LINKED ISSUE #187: Add bun support
I develop my website using bun as my package manager. Unfortunately the only supported package managers in this action are npm, pnpm, and yarn. I would like to add bun as a supported package manager via PR.

(I've already written the PR, but read the contributing guidelines a bit late and noticed you guys want an issue first. If you end up declining the PR, no sweat, my fault.)

### PR DESCRIPTION
**Pull request by simpleauthority**
This PR adds [Bun](https://bun.sh/) support and closes #187.

I wrote the PR before I read the contributing guidelines (sorry) and did not make an issue before I started working. As such, if you decline this it's not a big deal.

However, worth noting, it does work and I can deploy my website to CF pages with this.

### CODE REVIEW

**Comment by 1000hz** [comment]
Hey @simpleauthority! Happy to accept this PR, but I just merged #190 which refactored some of the package manager stuff. Could you please rebase and port your changes over? All of the package manager logic is now in `packageManagers.ts`.

Also, please run `bunx changeset` to add a `minor` version bump. Thanks for opening this PR!

**Comment by simpleauthority** [comment]
Sure thing, I will take a look.

**Comment by simpleauthority** [comment]
Rebased, tests all pass, was able to deploy my website with this, and added the changeset. Let me know if you need anything else @1000hz. Thanks!

File: test/bun/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "wrangler-action-npm-test",
```
Comment (1000hz) [inline test/bun/package.json:2]:
```suggestion
  "name": "wrangler-action-bun-test",
```

**Review by 1000hz — APPROVED** [review]

**Comment by 1000hz** [comment]
Oh, one more thing I just realized before I merge. Could you add `bun` as an option to the documentation in [action.yml](https://github.com/cloudflare/wrangler-action/blob/main/action.yml#L45)?

**Comment by simpleauthority** [comment]
Done!

**Review by 1000hz — APPROVED** [review]

**Pull request merged**